wasatch/OceanDevice.py

```diff
@@ -0,0 +1,181 @@
+            # issues sometimes that cseabreeze doesnt
+            if device.model == self.device_id.product and device.serial_number == self.device_id.serial:
+                self.device = device
+        if self.device == None:
+            log.error("Ocean Device: No ocean device found. Returning")
+            self.message_queue.put_nowait(None)
+            return False
+        self.spec = Spectrometer(self.device)
+        self.settings.eeprom.model = self.device.model
+        self.settings.eeprom.serial_number = self.device.serial_number
+        rev = self.spec.features["revision"][0]
+        self.settings.microcontroller_firmware_version = str(rev.revision_firmware_get())
+        self.settings.eeprom.detector = "Ocean" # Ocean API doesn't have access to detector info
+        return True
+
+
+    def init_lambdas(self):
+        f = {}
+        f["integration_time_ms"] = lambda x: self.spec.integration_time_micros(int(round(x)))
+        self.lambdas = f
+
+    def acquire_data(self):
+        self.settings.wavelengths = self.spec.wavelengths()# setting wavelengths one init doesn't work for some reaons
+        reading = self.take_one_averaged_reading()
+        return reading
+
+    def take_one_averaged_reading(self):
+        averaging_enabled = (self.settings.state.scans_to_average > 1)
+
+        if averaging_enabled and not self.settings.state.free_running_mode:
+            # collect the entire averaged spectrum at once (added for
+            # BatchCollection with laser delay)
+            #
+            # So: we're NOT in "free-running" mode, so we're basically being
+            # slaved to parent process and doing exactly what is requested
+            # "on command."  That means we can perform a big, heavy blocking
+            # scan average all at once, because they requested it.
+            self.sum_count = 0
+            loop_count = self.settings.state.scans_to_average
+        else:
+            # we're in free-running mode
+            loop_count = 1
+
+        log.debug("take_one_averaged_reading: loop_count = %d", loop_count)
+
+        # either take one measurement (normal), or a bunch (blocking averaging)
+        reading = None
+        for loop_index in range(0, loop_count):
+
+            # start a new reading
+            # NOTE: reading.timestamp is when reading STARTED, not FINISHED!
+            reading = Reading(self.device_id)
+
+            # TODO...just include a copy of SpectrometerState? something to think
+            # about. That would actually provide a reason to roll all the
+            # temperature etc readouts into the SpectrometerState class...
+            reading.integration_time_ms = self.settings.state.integration_time_ms
+            reading.laser_power_perc    = self.settings.state.laser_power_perc
+            reading.laser_power_mW      = self.settings.state.laser_power_mW
+            reading.laser_enabled       = self.settings.state.laser_enabled
```

Comment:
I'd be tempted to just delete all the laser stuff (leave them at Reading() defaults).